test/standalone/test-debuglet.js

```diff
@@ -95,6 +96,39 @@ describe(__filename, function(){
     debuglet.start();
   });
 
+  it('should pass source context to api if present', function(done) {
+    fs.writeFileSync('source-context.json', '{ "a": 5 }', 'utf8');
+    var debuglet = new Debuglet(
+      config, logger.create(config.logLevel, '@google/cloud-debug'));
+
+    process.env.GCLOUD_PROJECT_NUM=0;
+
+    var API = 'https://clouddebugger.googleapis.com';
+    var PATH = '/v2/controller/debuggees/register';
+    var scope = nock(API)
+      .post(PATH, function(body) {
+        return body.debuggee.sourceContexts[0].a === 5;
+      })
+      .reply(200, {
+        debuggee: {
+          id: 'foo'
+        }
+      });
+
+    debuglet.once('started', function() {
+      debuglet.debugletApi_.request_ = request; // Avoid authing.
+    });
+    debuglet.once('registered', function(id) {
+      assert(id === 'foo');
+      debuglet.stop();
+      scope.done();
+      fs.unlinkSync('source-context.json');
+      done();
+    });
```

Comment:
This pattern of code is showing up with regularity. Do you think it is worth refactoring this?